src/main/java/bio/terra/service/auth/iam/IamAction.java

```diff
@@ -41,6 +44,10 @@ public enum IamAction {
     this.samActionName = name().toLowerCase();
```

Comment:
When you have more than one constructer, I think it's usually better to chain constructors when possible (going from ones with fewer arguments to ones with more arguments), as I find it's easier to see how the different constructors relate to each other. In this case:
```suggestion
    this(name().toLowerCase());
```

Reply:
I like this, that makes sense.

Reply:
Actually, this does not work for our enum construction:

```
Cannot reference 'Enum.name' before supertype constructor has been called
```

Reply:
Oh yeah, was I was typing this I was wondering if that would work, as `enum` construction doesn't allow a few things that you can do elsewhere. Oh well.